crates/re_types/definitions/rerun/archetypes/boxes2d.fbs

```diff
@@ -0,0 +1,75 @@
+include "fbs/attributes.fbs";
+
+include "rerun/datatypes.fbs";
+include "rerun/components.fbs";
+
+namespace rerun.archetypes;
+
+// ---
+
+/// A batch of 2d boxes with half-extents and optional center, rotations, rotations, colors etc.
+///
+/// \py Example
+/// \py -------
+/// \py
+/// \py Simple 2D boxes:
+/// \py ```python
+/// \py \include:../../../../../docs/code-examples/box2d_simple.py
+/// \py ```
+///
+/// \rs ## Example
+/// \rs
+/// \rs Simple 2D boxes:
+/// \rs ```ignore
+/// \rs \include:../../../../../docs/code-examples/box2d_simple.rs
+/// \rs ```
+///
+/// \cpp ## Example
+/// \cpp
+/// \cpp Simple 2D boxes:
+/// \cpp ```ignore
+/// \cpp \include:../../../../../docs/code-examples/box2d_simple.cpp
+/// \cpp ```
+table Boxes2D (
+  "attr.rust.derive": "PartialEq",
+  order: 100
+) {
+  // --- Required ---
+
+  /// All half-extents that make up the batch of boxes.
+  half_extents: [rerun.components.HalfExtents2D] ("attr.rerun.component_required", order: 1000);
+
+  // --- Recommended ---
+
+  /// Optional center positions of the boxes.
+  centers: [rerun.components.Origin2D] ("attr.rerun.component_recommended", nullable, order: 2000);
```

Comment:
Making the centers optional seems weird to me. When would we want to have a bunch of sizes around the origin?

This is related to our `Arrow3D` archetype where the origin is also optional. This has lead to a very confusing example of arrows where we set directions, but no origins.

Reply:
In our tests and examples I have a bunch of cases where I log from the origin. I think it's nice that it is not *strictly required*, that is what the alternative would be.
Also, once we have 2D transforms, box2d without center will make even more sense

Reply:
Good point about the transform